pkg/assembler/backends/inmem/hasSBOM.go

```diff
@@ -82,11 +98,57 @@ func (c *demoClient) IngestHasSBOMs(ctx context.Context, subjects model.PackageO
 	return modelHasSboms, nil
 }
 
-func (c *demoClient) IngestHasSbom(ctx context.Context, subject model.PackageOrArtifactInput, input model.HasSBOMInputSpec) (*model.HasSbom, error) {
-	return c.ingestHasSbom(ctx, subject, input, true)
+func (c *demoClient) IngestHasSbom(ctx context.Context, subject model.PackageOrArtifactInput, input model.HasSBOMInputSpec, includes model.HasSBOMIncludesInputSpec) (*model.HasSbom, error) {
+	funcName := "IngestHasSbom"
+
+	var softwareIDs []uint32
+	var dependencyIDs []uint32
+	var occurrenceIDs []uint32
+	var err error
+
+	if includes.Software != nil {
+		if softwareIDs, err = parseIDs(includes.Software); err != nil {
+			return nil, gqlerror.Errorf("%v ::  %s", funcName, err)
+		}
+		for _, id := range softwareIDs {
+			node, ok := c.index[id]
+			if ok {
+				if _, ok = node.(*pkgVersionNode); !ok {
+					_, ok = node.(*artStruct)
+				}
+			}
+			if !ok {
+				return nil, gqlerror.Errorf("%v :: software id %d is neither an ingested Package nor an ingested Artifact", funcName, id)
+			}
+		}
+		softwareIDs = sortAndRemoveDups(softwareIDs)
+	}
+	if includes.Dependencies != nil {
+		if dependencyIDs, err = parseIDs(includes.Dependencies); err != nil {
+			return nil, gqlerror.Errorf("%v ::  %s", funcName, err)
+		}
+		for _, id := range dependencyIDs {
+			if isDependency, err := byID[*isDependencyLink](id, c); isDependency == nil || err != nil {
```

Comment:
hmm qq: why the extra check for `isDependency==nil` seems like this shouldn't be the case? if so there's probably a bug somewhere else?

Reply:
+1, I can remove it

Reply:
done